src/win/fs.c

```diff
@@ -94,7 +94,7 @@
 
 #define TIME_T_TO_FILETIME(time, filetime_ptr)                              \
   do {                                                                      \
-    uint64_t bigtime = ((int64_t) (time) * 10000000LL) +                    \
+    uint64_t bigtime = ((uint64_t) ((time) * 10000000LL)) +                 \
```

Comment:
If you multiply by `10000000ULL` (note the U in ULL), then you don't need an explicit cast to uint64_t.

Also, I confess I don't understand why it needs to be multiplied by 10 million.